specs/experimental/gov-token.md

```diff
@@ -0,0 +1,98 @@
+# Governance Token
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Hook-based Integration with `GovernanceDelegation`](#hook-based-integration-with-governancedelegation)
+  - [Token Minting](#token-minting)
+  - [Token Burning](#token-burning)
+  - [Voting Power](#voting-power)
+    - [Queries](#queries)
+  - [Delegation](#delegation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constants      | Value                                        |
+|----------------|----------------------------------------------|
+| Address        | `0x4200000000000000000000000000000000000042` |
+| Token name     | `Optimism`                                   |
+| Token symbol   | `OP`                                         |
+| Token decimals | `18`                                         |
+
+`GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
+`ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
+voting system. The contract integrates with the `GovernanceDelegation` contract through a hook-based approach to support
+advanced delegation.
+
+### Hook-based Integration with `GovernanceDelegation`
+
+Advanced delegation includes relative and absolute partial delegation, which allow delegators to distribute their voting
+power among multiple delegates in a fractional manner. The `_afterTokenTransfer` function in the `GovernanceToken` is
+modified to call the `afterTokenTransfer` function in the `GovernanceDelegation` contract, allowing the `GovernanceDelegation`
+contract to consume the hooks and update its delegation and checkpoint mappings accordingly.
+
+If the call to the `GovernanceDelegation`'s `afterTokenTransfer` function fails, the token transfer MUST revert. This ensures
+that the `GovernanceDelegation` contract remains in sync with the `GovernanceToken`. Otherwise, the `GovernanceToken` could
+be left in an inconsistent state relative to the `GovernanceDelegation` contract, such as when a token transfer is successful
+but the delegation state is not updated.
+
+All delegation-related state, including the `delegates`, `checkpoints`, and `numCheckpoints` mappings, is gradually
```

Comment:
`numCheckpoints` is not a mapping itself, but rather a getter. so we can write instead:

```suggestion
All delegation-related state, including the `_delegates` and `_checkpoints` mappings, and the `_totalSupplyCheckpoints` array, is gradually
```

to also include the total supply checkpoints